users/engler/fairseq/training.py

```diff
@@ -76,26 +101,37 @@ def __init__(
             self.keep_epochs = set(keep_epochs)
         self.max_epoch = max_epoch
         self.save_interval = save_interval
+        self.fairseq_python_exe = (
+            fairseq_python_exe
+            if fairseq_python_exe is not None
+            else gs.FAIRSEQ_PYTHON_EXE
```

Comment:
Actually it should be okay not to have a `fairseq_python_exe`, right? If `fairseq_hydra_exe` is given, this is not necessary, is it? If not, we should allow that it is `None` and `gs.FAIRSEQ_PYTHON_EXE` is not defined.

Then, if it actually is necessary later, we should assert that it is not `None`.
```suggestion
            else getattr(gs, "FAIRSEQ_PYTHON_EXE", None)
```

Reply:
Okay